addons/mail/static/src/components/file_uploader/file_uploader_tests.js

```diff
@@ -87,6 +89,48 @@ QUnit.test('no conflicts between file uploaders', async function (assert) {
     );
 });
 
+QUnit.test('does not crash when an attachment is removed before its upload starts', async function (assert) {
+    assert.expect(1);
+
+    const attachmentUploadedPromise = makeTestPromise();
+    await this.start({
+        async mockFetch(resource, init) {
+            const _super = this._super.bind(this, ...arguments);
+            if (resource === '/web/binary/upload_attachment') {
+                await attachmentUploadedPromise;
+            }
+            return _super();
+        },
+    });
+    const fileUploader = await this.createFileUploaderComponent();
+    const file1 = await createFile({
+        name: 'text1.txt',
+        content: 'hello, world',
+        contentType: 'text/plain',
+    });
+    const file2 = await createFile({
+        name: 'text2.txt',
+        content: 'hello, world',
+        contentType: 'text/plain',
+    });
+    inputFiles(
+        fileUploader.el.querySelector('.o_FileUploader_input'),
+        [file1, file2]
+    );
+    await nextAnimationFrame();
+    const attachment = this.env.models['mail.attachment'].find(attachment =>
+        attachment.isTemporary &&
+        attachment.filename === 'text2.txt'
+    );
+    await attachment.remove();
```

Comment:
Missing comment what's the intention here.

I wouldn't do that, to simulate attachment deletion by using model API. Instead I would explicitly make user interaction to delete the attachment, i.e. click on X button on `Attachment` component related to this attachment.